src/App/Pages/Settings/AutofillServicesPageViewModel.cs

```diff
@@ -176,8 +178,19 @@ public void ToggleInlineAutofill()
             InlineAutofillToggled = !InlineAutofillToggled;
         }
 
-        public void ToggleAccessibility()
+        public async Task ToggleAccessibilityAsync()
         {
+            if (!_deviceActionService.AutofillAccessibilityServiceRunning())
+            {
+                var accept = await _platformUtilsService.ShowDialogAsync(AppResources.AccessibilityDisclosureText,
+                    AppResources.AccessibilityDisclosureTitle, AppResources.AcceptAccessibilityDisclosure,
+                    AppResources.DeclineAccessibilityDisclosure);
+                if (accept)
+                {
+                    _deviceActionService.OpenAccessibilitySettings();
+                }
+                return;
```

Comment:
You can simplify it by:
```c#
if (!accept)
{
    return;
}
```
and if it's accepted it will just use the call outside the if to open the a11y settings